### PR TITLE
[BugFix] Handle file writer init error correctly

### PR DESCRIPTION
**Pull request by letian-jiang**
## What I'm doing:

Register file writer if it inits successfully.

Fixes https://github.com/StarRocks/StarRocksTest/issues/6741

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/connector_sink/hive_chunk_sink.cpp [hunk 1]
```diff
@@ -57,10 +57,11 @@ StatusOr<ConnectorChunkSink::Futures> HiveChunkSink::add(ChunkPtr chunk) {
...
 
     // create writer if not found
-    if (_partition_writers[partition] == nullptr) {
+    if (!_partition_writers.contains(partition)) {
         auto path = _partition_column_names.empty() ? _location_provider->get() : _location_provider->get(partition);
-        ASSIGN_OR_RETURN(_partition_writers[partition], _file_writer_factory->create(path));
-        RETURN_IF_ERROR(_partition_writers[partition]->init());
+        ASSIGN_OR_RETURN(auto writer, _file_writer_factory->create(path));
+        RETURN_IF_ERROR(writer->init());
+        _partition_writers[partition] = writer;
```
Comment (imay) [inline be/src/connector_sink/hive_chunk_sink.cpp:64]: You can use `_partition_writers.emplace(partition, std::move(writer))`

Comment (letian-jiang) [reply be/src/connector_sink/hive_chunk_sink.cpp]: get

**Comment by DorianZheng** [comment]
Please DRY, extract these common codes as a function or class

**Review by imay — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by letian-jiang** [comment]
@mergify backport branch-3.3

**Pull request merged**